service/frontend/workflow_handler.go

```diff
@@ -915,6 +943,15 @@ func (wh *WorkflowHandler) RespondWorkflowTaskCompleted(
 		return nil, errIdentityTooLong
 	}
 
+	if currentSpan := trace.SpanFromContext(ctx); currentSpan.IsRecording() {
+		if taskToken, err := wh.tokenSerializer.Deserialize(request.TaskToken); err == nil {
+			currentSpan.SetAttributes(
+				telemetry.WorkerIdentityAttr(request.Identity),
+				telemetry.WorkflowIdAttr(taskToken.WorkflowId),
+			)
+		}
+	}
```

Comment:
Wrapping this in a `IsRecording` (akin to a `IsDebug`) check due to the performance impact of the task token deserialization.